### PR TITLE
Prometheus new template rebase

### PR DESCRIPTION
**Pull request by mjudeikis**
Updating Prometheus for new templates/example.

1. New scraping rules, fixes
2. exposed alert manager
3. clean ansible
4. add a custom annotation for routes (in the example when AVI router in use we need to be able to add custom annotations)
5. Externalise some of the configs

Still work in progress...
FIY: @zgalor

### CODE REVIEW

**Comment by mjudeikis** [comment]
```
INSTALLER STATUS ***************************************************************************************************************************************************************                                   
Initialization             : Complete (0:00:15)     
Prometheus Install         : Complete (0:00:33)     

Monday 22 January 2018  15:03:22 +0000 (0:00:00.076)       0:00:47.301 ********                          
===============================================================================                          
openshift_prometheus : Set alert and prometheus secrets ----------------------------------------------------------------------------------------------------------------- 4.18s                                    openshift_prometheus : create prometheus and alerts routes -------------------------------------------------------------------------------------------------------------- 4.17s                                    openshift_prometheus : annotate serviceaccount -------------------------------------------------------------------------------------------------------------------------- 3.02s                                    openshift_prometheus : create services for prometheus ------------------------------------------------------------------------------------------------------------------- 2.84s                                    openshift_sanitize_inventory : Check for usage of deprecated variables -------------------------------------------------------------------------------------------------- 1.96s                                    openshift_prometheus : annotate prometheus service ---------------------------------------------------------------------------------------------------------------------- 1.84s                                    Gather Cluster facts ---------------------------------------------------------------------------------------------------------------------------------------------------- 1.64s                                    openshift_prometheus : Set prometheus stateful set ---------------------------------------------------------------------------------------------------------------------- 1.62s                                    include_role ------------------------------------------------------------------------------------------------------------------------------------------------------------ 1.51s                                    openshift_prometheus : Create templates subdirectory -------------------------------------------------------------------------------------------------------------------- 1.18s                                    openshift_prometheus : Set prometheus configmap ------------------------------------------------------------------------------------------------------------------------- 1.14s                                    Initialize openshift.node.sdn_mtu --------------------------------------------------------------------------------------------------------------------------------------- 1.10s                                    openshift_prometheus : Add prometheus project --------------------------------------------------------------------------------------------------------------------------- 0.98s                                    openshift_prometheus : Set prometheus template -------------------------------------------------------------------------------------------------------------------------- 0.97s                                    openshift_prometheus : template ----------------------------------------------------------------------------------------------------------------------------------------- 0.96s                                    openshift_prometheus : template ----------------------------------------------------------------------------------------------------------------------------------------- 0.95s                                    openshift_prometheus : template ----------------------------------------------------------------------------------------------------------------------------------------- 0.94s                                    openshift_prometheus : Set cluster-reader permissions for prometheus ---------------------------------------------------------------------------------------------------- 0.91s                                    openshift_prometheus : get oc version ----------------------------------------------------------------------------------------------------------------------------------- 0.88s                                    openshift_prometheus : create prometheus serviceaccount ----------------------------------------------------------------------------------------------------------------- 0.88s 
```

```
NAME           READY     STATUS    RESTARTS   AGE
prometheus-0   6/6       Running   0          1m
```

```
alertmanager   alertmanager-openshift-metrics.apps.192.168.20.160.xip.io             alertmanager   <all>     reencrypt     None
alerts         alerts-openshift-metrics.apps.192.168.20.160.xip.io                   alerts         <all>     reencrypt     None
prometheus     prometheus-openshift-metrics.apps.192.168.20.160.xip.io               prometheus     <all>     reencrypt     None
```

**Comment by mjudeikis** [comment]
@simonpasquier fiy. checking your HA repo too now to make appropriate changes to upstream example

File: roles/openshift_prometheus/templates/prometheus.j2 [hunk 1]
```diff
@@ -104,7 +104,7 @@ spec:
         - mountPath: /prometheus
           name: prometheus-data
 
-      # Deploy alertmanager behind prometheus-alert-buffer behind an oauth proxy
+      # Deploy alerts behind alerts-proxy behind an oauth proxy
```
Comment (zgalor) [inline roles/openshift_prometheus/templates/prometheus.j2]: `Deploy alert-buffer behind alerts-proxy` is clearer

File: roles/openshift_prometheus/templates/prometheus.j2 [hunk 2]
```diff
@@ -174,6 +174,48 @@ spec:
         - containerPort: 9099
           name: alert-buf
 
+      - name: alertmanager-proxy
```
Comment (zgalor) [inline roles/openshift_prometheus/templates/prometheus.j2]: Add comment `Deploy alertmanager behind alertmanager-proxy`

File: roles/openshift_prometheus/tasks/install_prometheus.yaml [hunk 3]
```diff
@@ -20,30 +22,23 @@
...
-    name: prometheus
+    name: "{{ openshift_prometheus_service_name }}"
     namespace: "{{ openshift_prometheus_namespace }}"
-    #    TODO add annotations when supproted
-    #    annotations:
-    #      serviceaccounts.openshift.io/oauth-redirectreference.prom: '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"prometheus"}}'
-    #      serviceaccounts.openshift.io/oauth-redirectreference.alerts: '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"alerts"}}'
-
-    secrets:
-      - prometheus-secrets
```
Comment (zgalor) [inline roles/openshift_prometheus/tasks/install_prometheus.yaml]: Where did this go?

**Comment by mjudeikis** [comment]
@sdodson who should review this? who owns Prometheus part?

File: roles/openshift_prometheus/templates/prometheus.yml.j2 [hunk 4]
```diff
@@ -1,10 +1,5 @@
 rule_files:
-  - 'prometheus.rules'
-{% if openshift_prometheus_additional_rules_file is defined and openshift_prometheus_additional_rules_file is not none %}
-  - 'prometheus.additional.rules'
-{% endif %}
-
-
+  - '*.rules'
```
Comment (zgalor) [inline roles/openshift_prometheus/templates/prometheus.yml.j2]: :+1: Nice!

**Comment by mjudeikis** [comment]
Thanks @zgalor. 
Checking now --mesh.peers thingy on our templates. Might try to put this one too into this. But mainly this is to let you guys know that Im doing this so we would not overlap :)

File: roles/openshift_prometheus/tasks/install_prometheus.yaml [hunk 5]
```diff
@@ -52,61 +47,52 @@
...
+      port: "{{ openshift_prometheus_service_port }}"
+      targetPort: "{{ openshift_prometheus_alerts_service_targetport }}"
+    - name: "{{ openshift_prometheus_alertmanager_service_name }}"
+      port: "{{ openshift_prometheus_service_port }}"
+      targetPort: "{{ openshift_prometheus_alertmanager_service_targetport }}"
+
+- name: create prometheus services obj
+  command: >
+    {{ openshift_client_binary }} apply -n {{ openshift_prometheus_namespace }}
+    -f {{ tempdir }}/svc/
```
[the remark this answers is not in the document]
Comment (mjudeikis) [reply roles/openshift_prometheus/tasks/install_prometheus.yaml]: If this is a case, I will replace. I have seen `oc_obj` working poorly with `svc` objects as they have immutable fields. If this was fixed I can move :)

File: roles/openshift_prometheus/tasks/install_prometheus.yaml [hunk 6]
```diff
@@ -52,61 +47,52 @@
...
-    service.alpha.openshift.io/serving-cert-secret-name=prometheus-tls
+    service {{ openshift_prometheus_service_name }} {{ item }}
+  with_items:
+    "{{ openshift_prometheus_service_annotations }}"
 
 - name: annotate alerts service
   command: >
     {{ openshift_client_binary }} annotate --overwrite -n {{ openshift_prometheus_namespace }}
-    service alerts 'service.alpha.openshift.io/serving-cert-secret-name=prometheus-alerts-tls'
+    service {{ openshift_prometheus_alerts_service_name }} {{ item }}
```
[the remark this answers is not in the document]
Comment (mjudeikis) [reply roles/openshift_prometheus/tasks/install_prometheus.yaml]: Work in progress... Need to do loop inside template, but "train trip was too short for this" :) will fix it

File: roles/openshift_prometheus/templates/prometheus.j2 (hunk 1 above)
Comment (mjudeikis) [reply roles/openshift_prometheus/templates/prometheus.j2]: ok

File: roles/openshift_prometheus/tasks/install_prometheus.yaml (hunk 3 above)
Comment (mjudeikis) [reply roles/openshift_prometheus/tasks/install_prometheus.yaml]: we created secret but never used it...
`Prometheus-secret` in tempalte in the end is Proxy secret from:
```
- name: Set alert and prometheus secrets
  oc_secret:
    state: present
    name: "{{ item }}-proxy"
    namespace: "{{ openshift_prometheus_namespace }}"
    contents:
      - path: session_secret
        data: "{{ 43 | lib_utils_oo_random_word }}="
  with_items:
    - prometheus
    - alerts
```

 think intention for this one was lost in development..

**Comment by mjudeikis** [comment]
@zgalor please review again :)

File: roles/openshift_prometheus/templates/prometheus.rules.j2 [hunk 7]
```diff
@@ -1,4 +1,4 @@
 groups:
 - name: example-rules
   interval: 30s # defaults to global interval
-  rules:
+  rules:
```
Comment (zgalor) [inline roles/openshift_prometheus/templates/prometheus.rules.j2]: add newline here

Comment (mjudeikis) [reply roles/openshift_prometheus/templates/prometheus.rules.j2]: I assume after? :D

**Pull request merged**